src/systemctl/systemctl.c

```diff
@@ -8213,24 +8213,29 @@ static int talk_initctl(void) {
         _cleanup_close_ int fd = -1;
         char rl;
         int r;
+        const char *init_fifo = "/run/initctl";
 
         rl = action_to_runlevel();
         if (!rl)
                 return 0;
 
         request.runlevel = rl;
 
-        fd = open(INIT_FIFO, O_WRONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY);
+        fd = open(init_fifo, O_WRONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY);
+        if (fd < 0 && errno == ENOENT) {
+                init_fifo = "/dev/initctl";
+                fd = open(init_fifo, O_WRONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY);
+        }
```

Comment:
I think it would be nicer like this:

```c
const char *p;
…
FOREACH_STRING(p, "/run/initctl", "/dev/initctl")  {
        fd = open(p, O_WRONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY);
        if (fd >= 0 || errno != ENOENT)
                break;
}
if (fd < 0) {
        if (errno == ENOENT)
                return 0;
        return log_error_errno(errno, …
…
```